domain/schema/model/sql/0018-machine.sql

```diff
@@ -58,6 +58,14 @@ CREATE TABLE machine_agent_version (
     REFERENCES architecture (id)
 );
 
+CREATE VIEW v_machine_target_agent_version AS
```

Comment:
What are you gaining by these views? They are only used once each. Just put the query in those places.

As to the queries themselves, have you costed this approach versus an unqualified join? I believe the join will be faster.

Reply:
Unqualified joins very slightly more efficient , so changed to that.